emcc.py

```diff
@@ -2693,33 +2707,40 @@ def modularize():
   logger.debug('Modularizing, assigning to var ' + shared.Settings.EXPORT_NAME)
   src = open(final).read()
 
+  # TODO: exports object generation for MINIMAL_RUNTIME
+  exports_object = '{}' if shared.Settings.MINIMAL_RUNTIME else shared.Settings.EXPORT_NAME
```

Comment:
I don't understand, without this it doesn't work at all, does it? The outside receives nothing?

Reply:
That's right. Like the TODO states, exports object is not yet implemented in MINIMAL_RUNTIME. Note that returning EXPORT_NAME here would not work for MINIMAL_RUNTIME, since MINIMAL_RUNTIME does not export all functions on Module like traditional runtime does (Module does not exist in MINIMAL_RUNTIME)

The desired solution here that the TODO refers to will be to implement a compile time mechanism for user to declare the specific exports they want out, instead of always outputting them all. That is, we do not distinguish between private exports (asm module -> js library code exports that can be fully minified) vs public exports (asm module -> user js code, or js library code -> user js code, that need to be unminified at export boundary; but still accessed as minified internally)

Reply:
What are `TODO states`?

I think I see, so is this PR just a step in the right direction, but it's not really usable yet? Or are you thinking it's usable even without exports (in the MODULARIZE_INSTANCE case)?

Reply:
>> Like the TODO states,

> What are `TODO states`?

Sorry, I meant, "like the TODO comment there refers to, ..."

> I think I see, so is this PR just a step in the right direction, but it's not really usable yet?

Yeah, the code just prepares towards the future scenario where instead of returning an empty exports object `{}`, we will actually return a proper object with populated exports there.